compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/internal/ClipboardUtils.desktop.kt

```diff
@@ -16,44 +16,66 @@
 
 package androidx.compose.foundation.internal
 
+import androidx.annotation.VisibleForTesting
 import androidx.compose.ui.platform.ClipEntry
 import androidx.compose.ui.platform.NativeClipboard
 import androidx.compose.ui.text.AnnotatedString
+import java.awt.datatransfer.Clipboard
+import java.awt.datatransfer.ClipboardOwner
 import java.awt.datatransfer.DataFlavor
-import java.awt.datatransfer.StringSelection
+import java.awt.datatransfer.Transferable
+import java.awt.datatransfer.UnsupportedFlavorException
 import java.io.IOException
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.withContext
+
+private val _annotatedStringFlavor: DataFlavor =
+    DataFlavor(AnnotatedString::class.java, "AnnotatedString")
+
+@Suppress("UnusedReceiverParameter")
+val DataFlavor.annotatedStringFlavor: DataFlavor
+    get() = _annotatedStringFlavor
 
 internal actual suspend fun ClipEntry.readText(): String? {
     if (!hasText()) return null
 
-    return withContext(Dispatchers.IO) {
-        val transferable = nativeClipEntry as? java.awt.datatransfer.Transferable
-        try {
-            transferable?.getTransferData(DataFlavor.stringFlavor) as? String
-        } catch (_: IOException) {
-            // the data is no longer available in the requested flavor
-            null
-        }
+    val transferable = nativeClipEntry as? Transferable
+    return try {
+        transferable?.getTransferData(DataFlavor.stringFlavor) as? String
+    } catch (_: IOException) {
+        // the data is no longer available in the requested flavor
+        null
     }
 }
 
-// TODO: https://youtrack.jetbrains.com/issue/CMP-7543/Support-styled-text-in-PlatformClipboard-implementations
 internal actual suspend fun ClipEntry.readAnnotatedString(): AnnotatedString? {
-    return readText()?.let { AnnotatedString(it) }
+    if (!hasAnnotatedString()) {
```

Comment:
This check is done to retain the existing behaviour in `androidx.compose.foundation.text.selection.TextFieldSelectionManager#paste`, which only tries to get an `AnnotatedString`:

```kotlin
val text = clipboard?.getClipEntry()?.readAnnotatedString() ?: return@launch
```

I will propose to Jetpack Compose to alter this slightly as:

```kotlin
val text = clipboard?.getClipEntry()?.readAnnotatedString()
  ?: clipboard?.getClipEntry()?.readText()?.let { AnnotatedString(it) }
  ?: return@launch
```

To make it more explicit at the usage site, but for now this covers that case.